src/main.rs

```diff
@@ -1665,16 +1634,98 @@ fn foreign_packages<'a>(
     metadata
         .packages
         .iter()
-        .filter(|package| package.is_third_party(&config.audit_as_crates_io))
+        .filter(|package| package.is_third_party(&config.policy))
 }
 
-fn first_party_packages<'a>(
+/// All first-party packages, **without** the audit-as-crates-io policy applied
+/// (because it's used for validating that field's value).
+fn first_party_packages_strict<'a>(
     metadata: &'a Metadata,
     config: &'a ConfigFile,
 ) -> impl Iterator<Item = &'a Package> + 'a {
     // Only analyze things from crates.io (no source = path-dep / workspace-member)
     metadata
         .packages
         .iter()
-        .filter(|package| !package.is_third_party(&config.audit_as_crates_io))
+        .filter(|package| !package.is_third_party(&config.policy))
+}
```

Comment:
The comment suggests that this is supposed to ignore audit_as_crates_io, but it calls the `is_third_party` method, meaning it actually would take `audit_as_crates_io` into account, no?

Perhaps this method should pass an empty `&SortedMap::new()` in instead of the actual policy table?

It would be nice to have some tests for `check_audit_as_crates_io` to catch issues like this.

Reply:
Good catch, fixed.